web-ng/src/app/components/main-page/main-page.component.ts

```diff
@@ -48,9 +52,30 @@ export class MainPageComponent implements OnInit {
         this.projectService.activateProject(params.get('projectId')!);
       })
     );
+    this.subscription.add(
+      this.route.fragment.subscribe(fragment => {
+        this.onFragmentChange(fragment);
+      })
+    );
   }
 
   ngOnDestroy() {
     this.subscription.unsubscribe();
   }
+
+  private onFragmentChange(fragment?: string) {
+    if (!fragment) {
+      return;
+    }
+    const params = new URLSearchParams(fragment);
+    if (params.get('l')) {
```

Comment:
where does this 'l' come from?

Reply:
Added a comment to clarify: "The 'l' param is used to represent the layer id being edited."

We should probably encapsulate these by wrapping the URL state and router in our objects, but I'll save that for future work for now.